File: py_validation/pardox/_bindings/compute.py
def bind(lib, c_void_p, c_char_p, c_int32, c_double, c_longlong, c_size_t, c_int64, c_int16):
    try:
        if hasattr(lib, 'pardox_filter_compare'):
            lib.pardox_filter_compare.argtypes = [c_void_p, c_char_p, c_void_p, c_char_p, c_int32]
            lib.pardox_filter_compare.restype = c_void_p

        if hasattr(lib, 'pardox_filter_compare_scalar'):
            lib.pardox_filter_compare_scalar.argtypes = [
                c_void_p, c_char_p,
                c_double, c_longlong, c_int32,
                c_int32
            ]
            lib.pardox_filter_compare_scalar.restype = c_void_p
    except Exception:
        pass

    try:
        if hasattr(lib, 'pardox_apply_filter'):
            lib.pardox_apply_filter.argtypes = [c_void_p, c_void_p, c_char_p]
            lib.pardox_apply_filter.restype = c_void_p
    except Exception:
        pass

    try:
        for fn in ['pardox_agg_sum', 'pardox_agg_mean', 'pardox_agg_min',
                   'pardox_agg_max', 'pardox_agg_count', 'pardox_agg_std']:
            if hasattr(lib, fn):
                getattr(lib, fn).argtypes = [c_void_p, c_char_p]
                getattr(lib, fn).restype = c_double
    except Exception:
        pass

    for fn, args, ret in [
        ('pardox_series_add', [c_void_p, c_char_p, c_void_p, c_char_p], c_void_p),
        ('pardox_series_sub', [c_void_p, c_char_p, c_void_p, c_char_p], c_void_p),
        ('pardox_series_mul', [c_void_p, c_char_p, c_void_p, c_char_p], c_void_p),
        ('pardox_series_div', [c_void_p, c_char_p, c_void_p, c_char_p], c_void_p),
        ('pardox_series_mod', [c_void_p, c_char_p, c_void_p, c_char_p], c_void_p),
    ]:
        if hasattr(lib, fn):
            getattr(lib, fn).argtypes = args
            getattr(lib, fn).restype  = ret

File: js_validation/pardox/src/_bindings/compute.js
'use strict';

module.exports = function bindCompute(_lib) {
    _lib.pardox_filter_compare = _lib.raw.func(
        'void * pardox_filter_compare(void * left, const char * left_col, void * right, const char * right_col, int op_code)'
    );
    _lib.pardox_filter_compare_scalar = _lib.raw.func(
        'void * pardox_filter_compare_scalar(void * mgr, const char * col, double val_f64, int64_t val_i64, int is_float, int op_code)'
    );
    _lib.pardox_apply_filter = _lib.raw.func('void * pardox_apply_filter(void * data, void * mask, const char * mask_col)');

    _lib.pardox_agg_sum   = _lib.raw.func('double pardox_agg_sum(void * mgr, const char * col)');
    _lib.pardox_agg_mean  = _lib.raw.func('double pardox_agg_mean(void * mgr, const char * col)');
    _lib.pardox_agg_min   = _lib.raw.func('double pardox_agg_min(void * mgr, const char * col)');
    _lib.pardox_agg_max   = _lib.raw.func('double pardox_agg_max(void * mgr, const char * col)');
    _lib.pardox_agg_count = _lib.raw.func('double pardox_agg_count(void * mgr, const char * col)');
    _lib.pardox_agg_std   = _lib.raw.func('double pardox_agg_std(void * mgr, const char * col)');

    _lib.pardox_series_add = _lib.raw.func('void * pardox_series_add(void * left, const char * left_col, void * right, const char * right_col)');
    _lib.pardox_series_sub = _lib.raw.func('void * pardox_series_sub(void * left, const char * left_col, void * right, const char * right_col)');
    _lib.pardox_series_mul = _lib.raw.func('void * pardox_series_mul(void * left, const char * left_col, void * right, const char * right_col)');
    _lib.pardox_series_div = _lib.raw.func('void * pardox_series_div(void * left, const char * left_col, void * right, const char * right_col)');
    _lib.pardox_series_mod = _lib.raw.func('void * pardox_series_mod(void * left, const char * left_col, void * right, const char * right_col)');
};
